dataline-analytics/src/main/java/io/dataline/analytics/TrackingClientSingleton.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Dataline
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.dataline.analytics;
+
+import io.dataline.config.ConfigSchema;
+import io.dataline.config.Configs;
+import io.dataline.config.EnvConfigs;
+import io.dataline.config.StandardWorkspace;
+import io.dataline.config.persistence.ConfigNotFoundException;
+import io.dataline.config.persistence.ConfigPersistence;
+import io.dataline.config.persistence.DefaultConfigPersistence;
+import io.dataline.config.persistence.JsonValidationException;
+import io.dataline.config.persistence.PersistenceConstants;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.function.Supplier;
+
+public class TrackingClientSingleton {
+
+  private static TrackingClient trackingClient;
+
+  public static TrackingClient get() {
```

Comment:
I'd like to avoid having to dependency inject the tracker everywhere. Would be curious if people are okay with this approach for tracking.

Reply:
I'm fine with the production tracker being a singleton but it seems like it needs to be injectable if we're going to test our metrics? And if we don't test our metrics we can't really trust them 100%.

Reply:
per our conversation offline, keeping this as a singleton is valuable so that we don't have to dependency inject everywhere. but in cases where we do need to test the behavior, we might dependency inject more locally into a class.

Reply:
@michel-tricot - i took a hack at this. not quite what you described. lmk if this works for you.

Reply:
Let's talk about it offline. I am not fan that you are creating a configpersistence that could be different from the rest of the application.

We shouldn't be afraid to have a static initialializer that needs to be called in the main (so you can inject options). Having the initializer would probably also solve the TrackingIdentiy supplier piece

Reply:
done.